operator/internal/controller/pulumi/stack_controller.go

```diff
@@ -1016,6 +1002,35 @@ func (sess *StackReconcilerSession) resolveResourceRefAsConfigItem(ctx context.C
 	}
 }
 
+// resolveResourceRef reads a referenced object and returns its value as a string.
+func (sess *StackReconcilerSession) resolveResourceRef(ctx context.Context, ref *shared.ResourceRef) (string, error) {
```

Comment:
Hopefully we can remove this method, and defer resolution of the value to the workspace pod.  In other words the operator shouldn't see the resolved value.

Reply:
Discussed offline - operator needs to resolve these in order to poll the remote for latest commit.